### ISSUE TITLE
Integer input, as a string, gives floating point output

### ISSUE THREAD

**Issue by gwerbin** [savoirfairelinux/num2words#182]
### Expected Behaviour

```
assert num2words('18') == 'eighteen
assert num2words(18) == 'eighteen'
```

### Actual Behaviour

```python
assert num2words('18') == 'eighteen point zero'
assert num2words(18) == 'eighteen'
```

### Steps to reproduce

```
$ python --version
Python 3.6.6 :: Anaconda, Inc.
$ pip install -y num2words
$ python -c 'from num2words import num2words; print(num2words("18")); print(num2words(18))'
```

**Comment by shulcsm** [comment]
I know this will break a lot of code, but shouldn't accepted types look like this:
```python
def to_cardinal(value: Union[int, float, Decimal]) -> str:
    pass

def to_ordinal(value: int) -> str:
    pass

def to_year(value: int) -> str:
    pass

def to_currency(value: Decimal) -> str:
    pass
```

At the moment input parsing is all over the place, IMO it should be responsibility of user

**Comment by erozqba** [comment]
I will only change this on your suggestion @shulcsm 
```
def to_currency(value: Union[int, float, Decimal]) -> str:
    pass
```

**Comment by shulcsm** [comment]
@erozqba can you rephrase that? Not sure i got it. Do you think it's acceptable as breaking change? If so we should probably figure out strict definitions for everything  considered "public API"

**Comment by gwerbin** [comment]
You could also go the [YARL](https://github.com/aio-libs/yarl) route and make a kind of `NumericThing` (obviously needs a better name) class, with methods like `NumericThing.as_currency()`

**Comment by erozqba** [comment]
@shulcsm I think we could make this possible. We should advise people about this change and give the opportunity to fix it before we make the release. We will need to announce this change and throw a warning if someone is using the wrong type, after 2 releases we could make the breaking change real.

**Comment by shulcsm** [comment]
Okay, that makes sense. I guess implementing sane sing path for parsing should be done after that.

**Comment by btharper** [comment]
I ran into an issue that's either another symptom or closely related based on handling strings as floats when the call `int(value) == value` call fails in num2words/base.py when trying to test numbers that don't fit in a float (> 10**23), like `num2words(str(10**80))`

Converting strings to Decimal objects fixes the integer string case above and large string numbers as well, while ignoring string parsing issues such as thousand separators and erroring and leaving it to the standard library (but not supporting alternate separators).

Using `locale.atof(number, Decimal)` may allow the caller to explicitly set their locale and use those parsing rules, but if the caller knows what locale the text is in, they could likely transform it themselves too. This is the only functionality I see in the standard library for doing locale aware parsing and it requires changing the effective locale of the whole interpreter, and is called out as something that libraries shouldn't use since it affects everything and can't be well isolated.
